src/Lecture1.hs

```diff
@@ -104,7 +113,13 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: [Char] -> Int
+strSum str = go 0 (words str)
+    where
+        -- go :: Int -> [[Char]] -> Int
+        go result xs
+            | null xs   = result
+            | otherwise = go (result + read (head xs) :: Int) (tail xs)
```

Comment:
There's already a standard function `sum` to find the sum of all elements in the list.

Also, as you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:
```suggestion
            | otherwise = go (result + read (head xs)) (tail xs)
```

If you use standard functions `sum` and `map`, you can implement `strSum` a short one-liner 🙂